src/Simulation/Simulators/CommonNativeSimulator/DisplayableState.cs

```diff
@@ -0,0 +1,243 @@
+        ///     Represents a quantum state vector and all metadata needed to display
+        ///     that state vector.
+        /// </summary>
+        public class DisplayableState
+        {
+            private static readonly IComparer<string> ToIntComparer =
+                Comparer<string>.Create((label1, label2) =>
+                    Comparer<int>.Default.Compare(
+                        Int32.Parse(label1), Int32.Parse(label2)
+                    )
+                );
+
+            /// <summary>
+            ///     Metadata to be used when serializing to JSON, allowing code
+            ///     in other languages to determine what representation is used
+            ///     for this state.
+            /// </summary>
+            [JsonProperty("diagnostic_kind")]
+            private string DiagnosticKind => "state-vector";
+
+            /// <summary>
+            ///     The indexes of each qubit on which this state is defined, or
+            ///     <c>null</c> if these indexes are not known.
+            /// </summary>
+            [JsonProperty("qubit_ids")]
+            public IEnumerable<int>? QubitIds { get; set; }
+
+            /// <summary>
+            ///     The number of qubits on which this state is defined.
+            /// </summary>
+            [JsonProperty("n_qubits")]
+            public int NQubits { get; set; }
+
+            /// <remarks>
+            ///     These amplitudes represent the computational basis states
+            ///     labeled in little-endian order, as per the behavior of
+            ///     <see cref="Microsoft.Quantum.Simulation.Simulators.CommonNativeSimulator.StateDumper.Dump" />.
+            /// </remarks>
+            [JsonProperty("amplitudes")]
+            public IDictionary<int, Complex>? Amplitudes { get; set; }
+
+            /// <summary>
+            ///     An enumerable source of the significant amplitudes of this state
+            ///     vector and their labels.
+            /// </summary>
+            /// <param name="convention">
+            ///     The convention to be used in labeling each computational basis state.
+            /// </param>
+            /// <param name="truncateSmallAmplitudes">
+            ///     Whether to truncate small amplitudes.
+            /// </param>
+            /// <param name="truncationThreshold">
+            ///     If <paramref name="truncateSmallAmplitudes" /> is <c>true</c>,
+            ///     then amplitudes whose absolute value squared are below this
+            ///     threshold are suppressed.
+            /// </param>
+            public IEnumerable<(Complex, string)> SignificantAmplitudes(
+                BasisStateLabelingConvention convention,
+                bool truncateSmallAmplitudes, double truncationThreshold
+            ) =>
+                (
+                    truncateSmallAmplitudes
+                    ? Amplitudes
+                        .Select(idx_amplitude => idx_amplitude)
```

Comment:
I think this line may not be needed now that `Amplitudes` is a dictionary?

```suggestion
```

In any case, `idx_amplitude` is a `KeyValuePair` with the new definition of `Amplitudes`, such that it's no longer an index:

<img width="494" alt="image" src="https://user-images.githubusercontent.com/31516/149430281-a80ed310-5faa-49a8-ae6a-a2512bcaa46f.png">

[nit / code style: the name `idx_amplitude` is in `snake_case` but should be in `camelCase`.]